app/controllers/course/assessment/question/text_responses_controller.rb

```diff
@@ -24,9 +28,18 @@ def create
 
   def edit
     @question_assessment = load_question_assessment_for(@text_response_question)
+    build_at_least_one_group_one_point if @text_response_question.comprehension_question?
   end
 
   def update
+    if @text_response_question.comprehension_question?
+      @text_response_question.groups.map do |group|
+        group.points.map do |point|
+          point.solutions.map(&:solution_will_change!)
```

Comment:
Why do you need this? Can give steps to reproduce the error? It seems to work fine without it for me.

Reply:
This forces `solution` array in TextResponseComprehensionSolutions to be marked as dirty, so `solution_lemma` will always be re-generated when the question is saved.

Reply:
Won't the regeneration already happen when the `solution` field changes? In which cases will we have to do this manual triggering?

Reply:
Assuming a valid TextResponseComprehensionQuestion has been created.
Let's say I manually change the `solution_lemma` in Postgres, then edit and save the question on the front end (without modifying the `solution` fields), the `solution_lemma` will not be re-generated if the above code chunk is not present.

This link is somewhat dated but still pretty much explains what I'm trying to say above: http://paweljaniak.co.za/2013/07/28/rails-4-and-postgres-arrays/

Reply:
Since the `solution` array is unlikely to be very big, I think it is okay to skirt around the issue mentioned in the article by avoiding array mutation altogether. 

With regards to changes directly made to the DB, invoking `solution_will_change!` will not guard against the case where an admin changes `solution_lemma` just before a submission is graded. Since the case is highly unlikely and does not guard against the problem completely, I don't think the case for keeping this is very compelling.

Since the aim is to keep `solution_lemma` up-to-date with `solution` and `solution_lemma` is only used during grading time, would it be safer to only do the lemmatization during grading time? Perhaps caching the lemmas in `solution_lemma` can be though of a performance optimization step that can be done if the need arises?

If we end up keeping this, we should have a spec for it so that whoever is working on your code next understands why it is there for.